oriedita/src/main/java/oriedita/editor/handler/MouseHandlerAxiom5.java

```diff
@@ -0,0 +1,246 @@
+                return;
+            }
+        }
+
+        // index 3 and 4 are the purple indicators
+        // 4. destination line (case 2)
+        if(d.getLineStep().size() == 5){
+            LineSegment closestLineSegment = new LineSegment();
+            closestLineSegment.set(d.getClosestLineSegment(p));
+
+            if (!(OritaCalc.determineLineSegmentDistance(p, closestLineSegment) < d.getSelectionDistance())) {
+                return;
+            }
+
+            closestLineSegment.setColor(LineColor.GREEN_6);
+            d.lineStepAdd(closestLineSegment);
+        }
+    }
+
+    @Override
+    public void mouseDragged(Point p0) {}
+
+    @Override
+    public void mouseReleased(Point p0) {
+        Point p = new Point();
+        p.set(d.getCamera().TV2object(p0));
+
+        // first 3 are clicked
+        if(d.getLineStep().size() == 3){
+            double radius = OritaCalc.distance(d.getLineStep().get(0).getA(), d.getLineStep().get(2).getA());
+            drawAxiom5FoldIndicators(radius, d.getLineStep().get(0).getA(), d.getLineStep().get(1), d.getLineStep().get(2).getA());
+        }
+
+        // Case 1: Click on the purple indicators auto expand the bisector from the purple indicators to the nearest lines
+        // (Kinda works)
+        if(d.getLineStep().size() == 5 && d.getClosestPoint(p).distance(p) > d.getSelectionDistance()){
+            if (OritaCalc.determineLineSegmentDistance(p, d.getLineStep().get(3)) < d.getSelectionDistance() ||
+                    OritaCalc.determineLineSegmentDistance(p, d.getLineStep().get(4)) < d.getSelectionDistance()) {
+                LineSegment s = d.get_moyori_step_lineSegment(p, 4, 5);
+                s.set(s.getB(), s.getA(), d.getLineColor());
+                s.set(OritaCalc.fullExtendUntilHit(d.getFoldLineSet(), s));
+
+                d.addLineSegment(s);
+                d.record();
+                d.getLineStep().clear();
+            }
+        }
+
+        // Case 2: Click on destination line and check of mouse point is closer to one of the 2 purple indicators
+        if(d.getLineStep().size() == 6){
+            Point intersectPoint1 = new Point(OritaCalc.findIntersection(d.getLineStep().get(3), d.getLineStep().get(5)));
+            Point intersectPoint2 = new Point(OritaCalc.findIntersection(d.getLineStep().get(4), d.getLineStep().get(5)));
+
+            double d1 = OritaCalc.distance(p, intersectPoint1);
+            double d2 = OritaCalc.distance(p, intersectPoint2);
+            d.addLineSegment(new LineSegment(d.getLineStep().get(2).getA(), d1 < d2 ? intersectPoint1 : intersectPoint2, d.getLineColor()));
+            d.record();
+            d.getLineStep().clear();
+        }
+    }
+    public void drawAxiom5FoldIndicators(double radius, Point target, LineSegment targetSegment, Point pivot) {
+        // Make sure circle radius is not 0
+        if(radius > Epsilon.UNKNOWN_1EN7){
```

Comment:
try to return early instead of having long if statements like this if possible, that allows you to stay at one indentation level instead of having to indent for every condition. in this case you can just do 
```java
if (radius <= Epsilon.UNKNOWN_1EN7) { return; }
// the rest of the code
```